.github/workflows/publish.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish new image version
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - "*"
+
+env:
+  REPO: quay.io/refgenomics/docker-onecodex-notebook
+
+jobs:
+  deploy:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Login to quay.io
+        run: echo ${{ secrets.QUAY_PASSWORD }} | docker login -u ${{ secrets.QUAY_USERNAME }} quay.io --password-stdin
+
+      - name: Pull latest image
+        run: docker pull ${{env.REPO}}:latest
+
+      - name: Extract branch/tag name
+        run: |
+          export TAG="${GITHUB_REF##*/}"
+          # Pushes to the "master" branch are published as "latest" image tag
+          if [ "$TAG" = "master" ]; then
+            export TAG="latest"
+          fi
+          echo "tag=$TAG" >> $GITHUB_ENV
+
+      - name: Build image
+        run: |
+          export DOCKER_BUILDKIT=1
+          docker build \
+            -t ${{env.REPO}}:${{env.tag}} \
+            --cache-from ${{env.REPO}} \
+            --build-arg BUILDKIT_INLINE_CACHE=1 \
+            .
```

Comment:
Do we need the cache there? Is the download + reuse of steps faster than building from scratch?

Reply:
It probably is, but you're right that it looks like a premature optimisation, let's add this when it's certain that it would speed up the process *significantly*